fusor-ember-cli/app/components/tr-hypervisor.js

```diff
@@ -30,26 +33,21 @@ export default Ember.Component.extend(TrEngineHypervisorMixin, {
     }
   ),
 
-  addOrRemoveHypervisor: Ember.observer('isSelectedAsHypervisor', function() {
-    if (this.get('isSelectedAsHypervisor')) {
-      this.get('model').addObject(this.get('host'));
-    } else {
-      this.get('model').removeObject(this.get('host'));
-    }
-  }),
-
-  setOrUnceckAll: Ember.observer('checkAll', 'uncheckAll', function() {
-    if (this.get('checkAll')) {
-      this.set('isSelectedAsHypervisor', true);
-    } else if (this.get('uncheckAll')) {
-      this.set('isSelectedAsHypervisor', false);
-    }
+  checkboxObserver: Ember.observer('isSelectedAsHypervisor', function() {
+    Ember.run.once(this, () => {
+      if (this.get('isSelectedAsHypervisor')) {
+        this.get('model').addObject(this.get('host'));
+      } else {
+        this.get('model').removeObject(this.get('host'));
+      }
+    });
   }),
 
-  isSelectedAsHypervisor: Ember.computed('selectedIds', 'host.id', function () {
-    if (this.get('selectedIds')) {
-      return this.get('selectedIds').contains(this.get('host.id'));
-    }
+  modelObserver: Ember.observer('model.[]', function() {
```

Comment:
Is there a cyclical observer dependency here?

checkboxObserver (dependent on isSelectedAsHypervisor) adds or removes objects to the model, modelObserver watches that array and sets isSelectedAsHypervisor, triggering checkboxObserver? Once loops break that?